src/coreclr/jit/importercalls.cpp

```diff
@@ -1303,11 +1303,10 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             eeGetCallSiteSig(pResolvedToken->token, pResolvedToken->tokenScope, pResolvedToken->tokenContext, sig);
         }
 
-        typeInfo tiRetVal = verMakeTypeInfo(sig->retType, sig->retTypeClass);
-
-        if (call->IsCall())
+        // Sometimes "call" is not a GT_CALL (if we imported an intrinsic that didn't turn into a call)
+        if (!bIntrinsicImported)
```

Comment:
Is this something that needs backported?

Looks like it's been a potential issue for a long time, although it's unclear to me why we're only just seeing this "now" given how widely used intrinsics have been, including in potentially tail recursive positions, and we've many that can be no-ops or just return one of the input operands (such as `op_UnaryPlus`, `AsInt32`, etc).

-- Notably looks like there are some asserts in this block checking that `!bIntrinsicImported` that can potentially be removed.

Reply:
Seems likely this double-processing ended up with the same result, so didn't actually cause a bug.

Reply:
I don't think it needs to be backported, I think it's very hard for the code that runs twice here to cause problems in release builds. Also we need a customer report for backporting.

> Looks like it's been a potential issue for a long time, although it's unclear to me why we're only just seeing this "now" given how widely used intrinsics have been, including in potentially tail recursive positions, and we've many that can be no-ops or just return one of the input operands (such as op_UnaryPlus, AsInt32, etc).

Presumably people do not write code that involves noop intrinsics like `op_UnaryPlus` very often, and even more rarely that also satisfies the other conditions to hit the bug.

> -- Notably looks like there are some asserts in this block checking that !bIntrinsicImported that can potentially be removed.

Let me address that as part of a future PR to rerunning CI.